fortex/health/processors/coreference_processor.py

```diff
@@ -0,0 +1,209 @@
+            (more greedy means more coreference links). Default: `0.5`.
+         - `max_dist` (`int`): How many mentions back to look when considering possible
+            antecedents of the current mention. Decreasing the value will cause
+            the system to run faster but less accurately. Default: `50`.
+         - `max_dist_match` (`int`): The system will consider linking the current mention
+            to a preceding one further than max_dist away if they share a noun or
+            proper noun. In this case, it looks max_dist_match away instead. Default: `500`.
+         - `blacklist` (`bool`): Should the system resolve coreferences for pronouns in the
+            following list: ["i", "me", "my", "you", "your"]. Default `True`.
+         - `store_scores` (`bool`): Should the system store the scores for the coreferences
+            in annotations. Default: `True`
+         - `conv_dict` (`dict(str, list(str))`): A conversion dictionary that you can use
+            to replace the embeddings of rare words (keys) by an average of the embeddings
+             of a list of common words (values). Ex: `conv_dict={"Angela": ["woman", "girl"]}`
+             will help resolving coreferences for Angela by using the embeddings for the more
+             common woman and girl instead of the embedding of Angela.
+             This currently only works for single words (not for words groups). Default: `None`.
+
+        Returns: A dictionary with the default config for this processor.
+        """
+        return {
+            "entry_type": "ft.onto.base_ontology.Document",
+            "mention_type": "ftx.medical.clinical_ontology.MedicalEntityMention",
+            "lang": "en_core_web_sm",
+            "model": True,
+            "greedyness": 0.5,
+            "max_dist": 50,
+            "max_dist_match": 500,
+            "blacklist": True,
+            "store_scores": True,
+            "conv_dict": None,
+        }
+
+    def expected_types_and_attributes(self):
+        r"""
+        Method to add user specified expected type which would be checked
+        before running the processor if the pipeline is initialized with
+        `enforce_consistency=True` or
+        :meth:`~forte.pipeline.Pipeline.enforce_consistency` was enabled for
+        the pipeline.
+        """
+        return {"ft.onto.base_ontology.Document": set()}
+
+    def record(self, record_meta: Dict[str, Set[str]]):
+        r"""
+        Method to add output type record of `CoreferenceProcessor` which
+        is `"ftx.medical.clinical_ontology.MedicalArticle"` with attribute
+        `coref_groups`, `has_coref`, `coref_scores`, and `coref_resolved`
+        to :attr:`forte.data.data_pack.Meta.record`.
+
+        Args:
+            record_meta: the field in the datapack for type record that need to
+                fill in for consistency checking.
+        """
+        record_meta["ftx.medical.clinical_ontology.MedicalArticle"] = {
```

Comment:
You haven't recorded the most important output: the `CoreferenceGroup` into this function.